frontend/frontend/src/screens/ListExpensesScreen.js

```diff
@@ -0,0 +1,199 @@
+                {
+                    amount: 0,
+                    date: today,
+                    description: 'choose a description',
+                    category: '617be036888f752511901458',
+                },
+                config
+            )
+
+            const dataCreated0 = {
+                _id: '1',
+            }
+            setCreatedExpense(data)
+        } catch (errors) {
+            setErrorCreate(
+                errors.response && errors.response.data.detail
+                    ? errors.response.data.detail
+                    : errors.message
+            )
+        } finally {
+            setLoadingCreate(false)
+        }
+    }
+
+    const deleteExpenseHandler = async (id) => {
+        if (window.confirm('Are you sure you want to delete this expense?')) {
+            try {
+                setLoadingDelete(true)
+                const config = {
+                    headers: {
+                        'Content-type': 'application/json',
+                    },
+                }
+                const { data } = await axios.delete(`http://localhost:5000/expenses/${id}`, config)
+                setDeletedExpense(data)
+            } catch (errors) {
+                setErrorDelete(
+                    errors.response && errors.response.data.detail
+                        ? errors.response.data.detail
+                        : errors.message
+                )
+            } finally {
+                setLoadingDelete(false)
+            }
+        }
+    }
+
+    useEffect(() => {
+        if (createdExpense) {
+            navigate(`/expense/${createdExpense._id}/edit`)
+            setCreatedExpense(null)
+        } else if (deletedExpense) {
+            setDeletedExpense(null)
+        }
+        getExpenses()
+    }, [navigate, createdExpense, deletedExpense])
+
+    return (
+        <div>
+            <Link to='/'>Go Back</Link>
+
+            <Row className='align-items-center'>
+                <Col>
+                    <h1>Expenses</h1>
+                </Col>
+
+                <Col className='text-right'>
+                    <div style={{ display: 'flex' }}>
+                        <Button
+                            className='my-3'
+                            style={{ marginLeft: 'auto' }}
+                            onClick={createExpenseHandler}
+                        >
+                            <i className='fas fa-plus'></i> Create Expense
+                        </Button>
+                    </div>
+                </Col>
+            </Row>
+
+            {loadingDelete && <Loader />}
+            {errorDelete && <Message variant='danger'>{errorDelete}</Message>}
+
+            {loadingCreate && <Loader />}
+            {errorCreate && <Message variant='danger'>{errorCreate}</Message>}
+
+            {loading ? (
+                <Loader />
+            ) : error ? (
+                <Message variant='danger'>{error}</Message>
+            ) : (
```

Comment:
Is there any way to organize all these Loaders and messages and make it more "DRY".

Reply:
There should be for sure, but I think that it is better to separate the delete, create and list loadings and messages here